portality/app.py

```diff
@@ -168,6 +168,16 @@ def utc_timestamp(stamp, string_format="%Y-%m-%dT%H:%M:%SZ"):
     utcdt = datetime(tt.tm_year, tt.tm_mon, tt.tm_mday, tt.tm_hour, tt.tm_min, tt.tm_sec, tzinfo=pytz.utc)
     return utcdt.strftime(string_format)
 
+@app.template_filter('doi_url')
+def doi_url(doi):
+    """
+    Create a link from a DOI.
+    :param doi: the string DOI
+    :return: the HTML link
+    """
+    tendot = doi[doi.find('10.'):]
+    return "<a href='http://dx.doi.org/{0}'>{0}</a>".format(tendot)
```

Comment:
I was replicating the following from facetview ```facetview_results_render_callbacks.js```, which I think drops anything before (but not including) the 10., in case the data has the http://... etc stuff in front of it:

``` javascript

fv_doi_link = (function (resultobj) {
    var that = function(resultobj) {
        if (resultobj.bibjson && resultobj.bibjson.identifier) {
            var ids = resultobj.bibjson.identifier
            for (var i = 0; i < ids.length; i++) {
                if (ids[i].type === "doi") {
                    var doi = ids[i].id
                    var tendot = doi.indexOf("10.")
                    var url = "http://dx.doi.org/" + doi.substring(tendot)
                    return "<a href='" + url + "'>" + doi.substring(tendot) + "</a>"
                }
            }
        }
        return false
    };
    return that;
})();
```